linux_kernel_modules/rtc_sync/rtc_sync.c

```diff
@@ -100,30 +102,32 @@ static int systohc(struct timespec now)
 	if (rtc) {
 		/* rtc_hctosys exclusively uses UTC, so we call set_time here,
 		 * not set_mmss. */
-		if (rtc->ops &&
-		    (rtc->ops->set_time ||
-		     rtc->ops->set_mmss))
+		if (rtc->ops && (rtc->ops->set_time || rtc->ops->set_mmss))
 			err = rtc_set_time(rtc, &tm);
-		rtc_class_close(rtc);
+			rtc_class_close(rtc);
```

Comment:
We need to call `rtc_class_close()` regardless of whether various ops are defined or not.